R/moduleInstall.R

```diff
@@ -1,12 +1,338 @@
+  if (!file.exists(pkgDescr))
+    stop("Your module contains no DESCRIPTION file")
+
+  return(read.dcf(file.path(modulePkg, "DESCRIPTION"))[1, ])
+
+}
+
+renv_diagnostics_packages_as_df <- function(project) {
+
+  lockfile  <- renv:::renv_diagnostics_packages_lockfile(project)
+  libstate  <- renv:::renv_diagnostics_packages_library(project)
+  used      <- unique(renv:::renv_diagnostics_packages_dependencies(project)$Package)
+  recdeps   <- renv:::renv_package_dependencies(packages = used, project = project)
+  all       <- c(names(lockfile$Packages), names(libstate$Packages),
+                 names(recdeps))
+
+  renv:::renv_scope_locale(category = "LC_COLLATE", locale = "C")
+  
+  all                        <- sort(unique(all))
+  deps                       <- rep.int(NA_character_, length(all))
+  names(deps)                <- all
+  deps[names(recdeps)]       <- "indirect"
+  deps[used]                 <- "direct"
+  libpaths                   <- dirname(renv:::map_chr(all, renv:::renv_package_find))
+  flibpaths                  <- factor(libpaths, levels = .libPaths())
+  libcodes                   <- as.integer(flibpaths)
+  libcodes[!is.na(libcodes)] <- sprintf("[%i]", libcodes[!is.na(libcodes)])
+
+  return(data.frame(
+    Library          = renv:::renv_diagnostics_packages_version(libstate, all),
+    Source           = renv:::renv_diagnostics_packages_sources(libstate, all),
+    Lockfile         = renv:::renv_diagnostics_packages_version(lockfile, all),
+    Source           = renv:::renv_diagnostics_packages_sources(lockfile, all),
+    Path             = libcodes,
+    Dependency       = deps,
+    stringsAsFactors = FALSE,
+    check.names      = FALSE
+  ))
+}
+
+libraryMatchesLockfile <- function(project = NULL) {
+  project <- renv:::renv_project_resolve(project)
+  df      <- renv_diagnostics_packages_as_df(project)
+  notNA   <- which(complete.cases(df[, c("Library", "Lockfile")]))
+  idxDiff <- which(df[notNA, "Library"] != df[notNA, "Lockfile"])
+  hasDiff <- length(idxDiff) > 0L
+
+  if (hasDiff) {
+    print("Found the following mismatches between Library and the lockfile!")
+    print(df[notNA[idxDiff], ])
+  }
+   
+  return(!hasDiff)
+}
+
+addRenvBeforeAfterDispatch <- function() { 
+
+  renBeforeAfterInstallStruct <- structure(list(
+    before.install = function(x) {
+      #print("BEFORE INSTALLING")
+      #print(sprintf("Path = %s", mget("path", envir = parent.frame(1),
+      #                                ifnotfound = "unknown path")))
+      0 #do nothing
+    }, 
+    
+    after.install = function(x) 
+    {
+      installPath <- mget("installpath", envir = parent.frame(1), ifnotfound = "unknown")
+
+      if(installPath != "unknown") 
+      {
+        print(sprintf("Installed %s to '%s', now running post install fixes.", x, installPath))
+        postInstallFixes(installPath)
+      } 
+      else
+        print(sprintf("Installing %s failed for some reason...", x))
```

Comment:
I think I've seen this message pop up a lot before `renv` decides to start looking at remotes, so it might be that the error msg is just a bit misguided.